src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogTcpClient.cs

```diff
@@ -136,11 +142,45 @@ private void CloseConnection()
 #endif
             _stream = null;
             _client = null;
+            _connectionMatcher = null;
         }
 
         private bool IsConnectionClosed()
         {
-            return _client == null || _stream == null;
+            if (_client == null || _stream == null)
+            {
+                return true;
+            }
+
+#if NETSTANDARD1_3
+            // `IPGlobalProperties` does not exist, keep the same behavior as before.
+            return false;
+#else
+            TcpConnectionInformation[] connections = null;
+            try
+            {
+                connections = IPGlobalProperties.GetIPGlobalProperties().GetActiveTcpConnections();
+            }
+            catch (NotImplementedException)
+            {
+                // Happen when using Mono on MacOs. Keep the same behavior as before.
+                return false;
+            }
+
+            if (_connectionMatcher != null)
+            {
+                var currentConnection = connections.FirstOrDefault(
+                    c => _connectionMatcher.IsSameConnection(c.LocalEndPoint, c.RemoteEndPoint));
```

Comment:
Trying to think of scenarios where `IsConnectionClosed` could cause false negatives (i.e. return `false` even if the connection is actually closed):

* I'm guessing connections from multiple processes on the host could match these conditions, for example if there are multiple .NET runtimes on the host, all using the Serilog plugin. Can you think of ways to improve this? (could we match on the process as well, or just get the connections of the current process?)
* Can 1 .NET process run multiple instances of the Serilog plugin? And can the Serilog plugin have more than one open connection? If any of these 2 cases can happen, then this logic could also match a connection of the same process that's actually not a connection held by the current plugin instance.

For both questions, if improving the logic you implemented is difficult/impossible, at least I think we should document these limitations in a comment on the method.

Reply:
As far as I know, `TCPClient` uses a unique and unused ports (system wide) each time the connexion is created.
The connection is defined by source IP, source port, destination IP, destination port.

I have also performed the following tests:
 - Create 2 instances of `DatadogSink` in the same process. Ports are different.
 - Create 1 instance of `DatadogSink` in two processes. Ports are different.

Reply:
Thanks.

I think there are socket options to re-use the same port across sockets, but seems not relevant in our case here.